### DISCUSSION TITLE
Using the required indicator when all input fields on a form are mandatory

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1108] [Office hours] [answered]
We're creating some new flows for upgrading an account. On some forms (such as the one in the wireframe below), all but one field is mandatory. In such cases, we would prefer to just mark the ones that aren't required as optional. 

As per the paste guidelines for [input fields ](https://paste.twilio.design/components/input/#accessibility), 
`Use the required prop to programatically indicate they are required to browsers.`

Do we have to use the 'required' prop on all the required fields, or will our option of using 'optional' on the ones that are suffice? Visually, we would prefer to not have a form where all field labels have a mandatory indicator on them.

![image](https://user-images.githubusercontent.com/72031986/105644787-5982a300-5e8f-11eb-96fa-f1b8579774ec.png)

**Comment by corinne-nagel** [comment, marked as answer]
Hi @PoojMenon , from a visual perspective, the [required indicator guidance](https://paste.twilio.design/components/input/#input) is a little lower on that page:
**Required field indicator** — In a form where there are fewer or as many required fields as optional, use a required indicator to show users which fields are required to be filled out.

If possible, researching this would be helpful to determine which direction is easiest for your users to complete the form (let us know if you do research this as it would be a great insight to have) but otherwise if **# required is < or equal to # optional** use a required indicator 👍

**Comment by PoojMenon** [comment]
Hi @corinne-nagel, Thank you for the response. That makes sense. We're going to be testing these soon. So will be sure to share if any insights we gather.
